controlplane/pkg/apis/local/networkservice/helpers.go

```diff
@@ -5,41 +5,82 @@ import (
 
 	"github.com/golang/protobuf/proto"
 
-	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
-	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/nsm"
+	local "github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/nsm/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/nsm/networkservice"
 )
 
+// NewRequest creates a new local.NetworkServiceRequest
+func NewRequest(connection connection.Connection, mechanismPreferences []connection.Mechanism) *NetworkServiceRequest {
+	ns := &NetworkServiceRequest{}
+
+	ns.SetRequestConnection(connection)
+	ns.SetRequestMechanismPreferences(mechanismPreferences)
+
+	return ns
+}
+
+// IsRemote returns if request is remote
+func (ns *NetworkServiceRequest) IsRemote() bool {
+	return false
+}
+
+// Clone clones request
+func (ns *NetworkServiceRequest) Clone() networkservice.Request {
+	return proto.Clone(ns).(*NetworkServiceRequest)
+}
+
+// GetRequestConnection returns request connection
+func (ns *NetworkServiceRequest) GetRequestConnection() connection.Connection {
+	return ns.GetConnection()
+}
+
+// SetRequestConnection sets request connection
+func (ns *NetworkServiceRequest) SetRequestConnection(connection connection.Connection) {
+	ns.Connection = connection.(*local.Connection)
+}
+
+// GetRequestMechanismPreferences returns request mechanism preferences
+func (ns *NetworkServiceRequest) GetRequestMechanismPreferences() []connection.Mechanism {
+	preferences := make([]connection.Mechanism, 0, len(ns.MechanismPreferences))
+	for _, m := range ns.MechanismPreferences {
+		preferences = append(preferences, m)
+	}
+
+	return preferences
+}
+
+// SetRequestMechanismPreferences sets request mechanism preferences
+func (ns *NetworkServiceRequest) SetRequestMechanismPreferences(mechanismPreferences []connection.Mechanism) {
+	preferences := make([]*local.Mechanism, 0, len(mechanismPreferences))
+	for _, m := range mechanismPreferences {
+		preferences = append(preferences, m.(*local.Mechanism))
```

Comment:
is missing check on type?

Reply:
`m.(*local.Mechanism)` will panic if `m` is not a `*local.Mechanism`. So it is actually the check.